### PR TITLE
Tests (+ Ember Upgrades / Restoring Ember 1.13.x Compatibility)

### LINKED ISSUE #5: Add tests
Since this addon dives into the router internals, it should have some meaningful tests. Probably acceptance tests using the dummy application since a lot of mocking would be required otherwise. The lack of tests are what's driving a 6/10 ember observer score.

### PR DESCRIPTION
**Pull request by blimmer**
Based off of PR #6 . This introduces some tests that should help get the Ember Observer score up, and provide meaningful tests. It closes #5 .

### CODE REVIEW

File: addon/services/route-layers.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ import Ember from 'ember';
 
 export default Ember.Service.extend({
 
-  stack: [],
+  stack: Ember.computed(function() {
+    return Ember.A([]);
+  }),
```
Comment (blimmer) [inline addon/services/route-layers.js:7]: This was the potential bug exposed by testing this addon. Before, we were defining the stack array on the prototype, which can throw `calling set on destroyed object` errors. By making this a simple computed property (so it's generated on the instance), it resolves the issue.

**Comment by blimmer** [comment]
We should turn on travis CI builds for this project now that we have a set of tests. [Here](https://travis-ci.org/blimmer/ember-route-layers/builds/113203965) are the tests passing on my fork.

EDIT: tests showed that 1.13.x compatibility was broken by [this commit](https://github.com/dollarshaveclub/ember-route-layers/commit/afc845ce1ac38d4ff06d114921156539a8e97e4a). I've made the updates to restore 1.13 functionality. [Tests passing here](https://travis-ci.org/blimmer/ember-route-layers/builds/113208753).

File: addon/initializers/route-layers.js [hunk 2]
```diff
@@ -1,43 +1,48 @@
...
-        var exitPoint = this.routeLayers.pop();
-        if (exitPoint) exitPoint.transition.retry();
-        else this.send('exitRouteLayerFallback');
-      },
-
-      exitRouteLayerFallback: function () {
-        this.transitionTo('index');
+  };
+
+  const actionHashName = EMBER_MAJOR_VERSION === '1' ? '_actions' : 'actions';
```
Comment (blimmer) [inline addon/initializers/route-layers.js:29]: This is what's needed for 1.13.x. Found the solution from rwjblue [here](https://github.com/emberjs/ember.js/issues/12743)

**Comment by arjansingh** [comment]
@blimmer Thanks for the submission. We'll review the code tomorrow and (hopefully) merge. :smile:

**Comment by blimmer** [comment]
Sounds good! I'm available on the ember community slack all day tomorrow if you'd like to chat.

**Comment by briangonzalez** [comment]
@blimmer I'll be around too tomorrow to help corral this one into master.

File: addon/initializers/route-layers.js (hunk 2 above)
Comment (briangonzalez) [inline addon/initializers/route-layers.js:29]: Simple and elegant. I like it.

**Comment by blimmer** [comment]
Awesome, thank you for your help in getting this to master. Talk to y'all later today.

**Comment by pwfisher** [comment]
@blimmer Damn, this is quality work. Thank you.

**Comment by blimmer** [comment]
No problem - thanks for merging. While you're on this train of thought, can I ask for some feedback on #8 ?

**Pull request merged**